Frends.AmazonS3.DownloadObject/Frends.AmazonS3.DownloadObject/Definitions/Connection.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace Frends.AmazonS3.DownloadObject
+{
+    /// <summary>
+    /// Connection parameters.
+    /// </summary>
+    public class Connection
+    {
+        /// <summary>
+        /// Authentication method to use when connecting to AWS S3 bucket. Options are pre-signed URL or AWS Access Key ID+AWS Secret Access Key.
+        /// </summary>
+        /// <example>AWSCredentials</example>
+        [DefaultValue(AuthenticationMethod.AWSCredentials)]
+        public AuthenticationMethod AuthenticationMethod { get; set; }
+
+        #region Pre-signed URL
+        /// <summary>
+        /// A pre-signed URL allows you to grant temporary access to users who don't have permission to directly run AWS operations in your account.
+        /// </summary>
+        /// <example>"https://bucket.s3.region.amazonaws.com/object/file.txt?X-Amz-Expires=120X-Amz-Algorithm...</example>
+        [PasswordPropertyText]
+        [UIHint(nameof(AuthenticationMethod), "", AuthenticationMethod.PreSignedURL)]
+        [DefaultValue(null)]
+        public string PreSignedURL { get; set; }
+        #endregion Pre-signed URL
+
+        #region AWSCredentials
```

Comment:
No need for region.

Reply:
It's easier (at least) for me to read the code when you can collapse pre-signed or AWS creds part out of sight in time to time. This doesn't affect the code.